cli/command/trust/signer_add.go

```diff
@@ -0,0 +1,100 @@
+package trust
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/trust"
+	"github.com/docker/notary"
+	"github.com/docker/notary/passphrase"
+	store "github.com/docker/notary/storage"
+	"github.com/docker/notary/tuf/data"
+	"github.com/docker/notary/utils"
+	"github.com/spf13/cobra"
+)
+
+type signerAddOptions struct {
+	key string
+}
+
+func newSignerAddCommand(dockerCli command.Cli) *cobra.Command {
+	options := signerAddOptions{}
+	cmd := &cobra.Command{
+		Use:   "signer-add [OPTIONS] IMAGE NAME",
+		Short: "Add a signer",
+		Args:  cli.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return addSigner(dockerCli, args[0], args[1], options)
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringVarP(&options.key, "key", "k", "", "Path to the signer key to add")
+	return cmd
+}
+
+func addSigner(cli command.Cli, imageName string, signerName string, options signerAddOptions) error {
+
+	// TODO@ashfall: instead of failing if no key is provided, ask user if
+	// they want to generate a key
+	if options.key == "" {
+		return fmt.Errorf("No key provided. Please use the --key flag to provide the path to a valid key")
+	}
+
+	_, ref, repoInfo, authConfig, err := getImageReferencesAndAuth(cli, imageName)
+	if err != nil {
+		return err
+	}
+
+	// TODO: initialize the repo if it's not already initialized with notary?
+	notaryRepo, err := trust.GetNotaryRepository(cli, repoInfo, *authConfig, "push", "pull")
+	if err != nil {
+		return trust.NotaryError(ref.Name(), err)
+	}
+
+	newSignerRoleName := data.RoleName(path.Join(data.CanonicalTargetsRole.String(), signerName))
+
+	passRetriever := passphrase.PromptRetrieverWithInOut(cli.In(), cli.Out(), nil)
+	trustDir := trust.GetTrustDirectory()
+	importers, err := getImporters(trustDir)
+	if err != nil {
+		return err
+	}
+	if err := importKeys(newSignerRoleName.String(), options.key, passRetriever, importers); err != nil {
+		return err
+	}
+
+	// Now retrieve the notary key to add signer
+	signerKey, err := getOrGenerateNotaryKey(notaryRepo, newSignerRoleName)
```

Comment:
@riyazdf Wondering if there's a way to eliminate this call altogether and have utils.ImportKeys give us the key as it adds?